### PR TITLE
RavenDB-7276 PR comments fixes (#2555):

### PR DESCRIPTION
**Pull request by arekpalinski**
- making sure that we dispose all ETLs on the db record change
- added missing ref
- prevent NRE

### CODE REVIEW

File: src/Raven.Server/Documents/ETL/EtlLoader.cs [hunk 1]
```diff
@@ -157,14 +158,32 @@ public virtual void Dispose()
         {
             _database.Changes.OnDocumentChange -= NotifyAboutWork;
 
-            Parallel.ForEach(_processes, x => x.Dispose());
+            var ea = new ExceptionAggregator(Logger, "Could not dispose ETL loader");
```
Comment (ayende) [inline src/Raven.Server/Documents/ETL/EtlLoader.cs:161]: The exception logger is not thread safe, IIRC

File: src/Raven.Server/Documents/ETL/EtlLoader.cs [hunk 2]
```diff
@@ -157,14 +158,32 @@ public virtual void Dispose()
...
-            Parallel.ForEach(old, x => x.Dispose());
+            Parallel.ForEach(old, x =>
+            {
+                try
+                {
+                    x.Dispose();
+                }
+                catch (Exception e)
+                {
+                    if (Logger.IsOperationsEnabled)
```
Comment (ayende) [inline src/Raven.Server/Documents/ETL/EtlLoader.cs]: Why is this operations level?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/ETL/EtlLoader.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/ETL/EtlLoader.cs:161]: Do you mean `ExceptionAggregator`? It uses `ConcurrentSet` under the hood so it should be fine

Comment (ayende) [reply src/Raven.Server/Documents/ETL/EtlLoader.cs:161]: Hm... was certain it was just a list, okay, great then

File: src/Raven.Server/Documents/ETL/EtlLoader.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/ETL/EtlLoader.cs]: Changed to info

**Pull request merged**